direnv/flake.nix

```
@@ -0,0 +1,15 @@
+{
+  description = "An empty devshell with direnv support";
+
+  inputs.nixpkgs.url = "github:auxolotl/nixpkgs/nixpkgs-unstable";
+
+  outputs =
+    { nixpkgs, ... }:
+    let
+      system = "x86_64-linux";
+      pkgs = nixpkgs.legacyPackages.${system};
+    in
+    {
+      devShells.${system}.default = pkgs.mkShell { packages = [ pkgs.hello ]; };
+    };
```

```suggestion
      forAllSystems =
        function:
        nixpkgs.lib.genAttrs [
          "x86_64-linux"
          "aarch64-linux"
          "x86_64-darwin"
          "aarch64-darwin"
        ] (system: function nixpkgs.legacyPackages.${system});
    in
    {
      devShells = forAllSystems (pkgs: {
        default = pkgs.mkShellNoCC {
          packages = [ pkgs.hello ];
          IN_SHELL = "yes!";
        };
      });
    };
```

we can follow our own flake here. `mkShellNoCC` should also be used as to avoid any massive download of toolchains. as for the env var, we could put the command `echo $IN_SHELL` in a tutorial